### PR TITLE
Bug #211487 [WEB UI] : Camp Session -> Unable to see the current session on top

### PR DESCRIPTION
**Pull request by chaitanyakole**
### I have ensured that following `Pull Request Checklist` is taken care of before sending this PR
* [ ] Code is formatted as per format decided
* [ ] Updated acceptance criteria in tracker
* [ ] Updated test cases in test-cases-tracker
https://tracker.tekdi.net/issues/211487

https://github.com/tekdi/eg-website/assets/124124878/43c1eb1e-22ab-4542-b470-0e728a73f194

## Summary by CodeRabbit

- **New Features**
	- Enhanced the Camp Session Modal to manage state for better user interaction.
	- Sessions are now filtered and displayed based on completion status for improved usability.

### CODE REVIEW

File: apps/front-end/src/pages/admin/camps/CampSessionModal.js [hunk 1]
```diff
@@ -153,6 +153,7 @@ const SessionActions = ({
         />
       ) : ( */}
       <VStack space={4}>
+        {console.log({ buttonHide })}
```
[the remark this answers is not in the document]
Comment (chaitanyakole) [reply apps/front-end/src/pages/admin/camps/CampSessionModal.js]: resolved

File: apps/front-end/src/pages/admin/camps/CampSessionModal.js [hunk 2]
```diff
@@ -226,9 +226,92 @@ export default React.memo(SessionActions);
 export const SessionList = React.memo(
   ({ sessionList, sessionActive, setModalVisible }) => {
     const { t } = useTranslation();
+    const [collapsed, setCollapsed] = useState(true);
+
+    // Separate completed and incomplete sessions
+    const completedSessions = sessionList.filter(
```
Comment (swapnilphalke03) [inline apps/front-end/src/pages/admin/camps/CampSessionModal.js:232]: add ?.filter

File: apps/front-end/src/pages/admin/camps/CampSessionModal.js [hunk 3]
```diff
@@ -226,9 +226,92 @@ export default React.memo(SessionActions);
...
+      (item) => item?.session_tracks?.[0]?.status === "complete"
+    );
+
+    const incompleteSessions = sessionList.filter(
```
Comment (swapnilphalke03) [inline apps/front-end/src/pages/admin/camps/CampSessionModal.js:236]: add ?.filter

File: apps/front-end/src/pages/admin/camps/CampSessionModal.js [hunk 4]
```diff
@@ -226,9 +226,92 @@ export default React.memo(SessionActions);
...
+                  <IconByName
+                    alignContent={"right"}
+                    name="ArrowRightSLineIcon"
+                    _icon={{ size: "25px" }}
+                  />
+                </HStack>
+              </CardComponent>
+            </Pressable>
+          ))}
+        {incompleteSessions.map((item) => (
```
Comment (swapnilphalke03) [inline apps/front-end/src/pages/admin/camps/CampSessionModal.js:314]: add ?.map

**Review by swapnilphalke03 — CHANGES_REQUESTED** [review]
check comments

File: apps/front-end/src/pages/admin/camps/CampSessionModal.js (hunk 2 above)
Comment (chaitanyakole) [reply apps/front-end/src/pages/admin/camps/CampSessionModal.js]: done

File: apps/front-end/src/pages/admin/camps/CampSessionModal.js (hunk 4 above)
Comment (chaitanyakole) [reply apps/front-end/src/pages/admin/camps/CampSessionModal.js:314]: Done

File: apps/front-end/src/pages/admin/camps/CampSessionModal.js (hunk 3 above)
Comment (chaitanyakole) [reply apps/front-end/src/pages/admin/camps/CampSessionModal.js]: done

**Review by swapnilphalke03 — APPROVED** [review]

**Pull request merged**